src/collection.js

```diff
@@ -365,10 +389,39 @@ export default class Collection {
         "max_target.data.last_modified": String(at),
       },
     });
+    return changes;
+  }
 
+  /**
+   * @private
+   */
+  async getSnapshot(at) {
+    if (!Number.isInteger(at) || at <= 0) {
+      throw new Error("Invalid argument, expected a positive integer.");
+    }
+    // First, check that the history plugin covers the entire range back to the
+    // requested timestamp.
+    // 1. find the record timestamp in the current collection which
+    // last_modified is immediately fresher than the requested timestamp.
+    const oldestRecord = await this.findOldestRecord(at);
+    if (!oldestRecord) {
+      // If the current records list is empty, we can't find its oldest
+      // record, so we can't ensure we have enough history data to compute
+      // an accurate snapshot, which is too risky.
+      throw new Error("Cannot compute a snapshot against an empty list.");
+    }
+    // 2. find the oldest history entry for the current collection
+    const oldestHistoryEntry = await this.findOldestHistoryEntry();
```

Comment:
Because `changes[0]` is not the oldest history entry, it's the oldest history entry matching the requested range. We need to ensure history has actually been enabled *before* the requested timestamp, which information is not provided by this query.